### PR TITLE
feature: fetching data from backend

### PR DESCRIPTION
**Pull request by nayyyhaa**
## In this PR :
-  I've just added backend integration for fetching products & categories data.
- added absolute path while importing files.
- NOTE: authorization is not been integrated yet.

### Files to check

- [x] [App.js](https://github.com/nayyyhaa/inNotion-care-react/commit/f4ae7c51a48f78a5589989e6d783da7134eff9c9#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)

#### Component files

- [x] [components/Card.js](https://github.com/nayyyhaa/inNotion-care-react/commit/f4ae7c51a48f78a5589989e6d783da7134eff9c9#diff-7916963985f6b54523951aa9bdcdb855db5905bed40fc7775a485986aec0fa5d)
- [x] [components/FeatureCategories.js](https://github.com/nayyyhaa/inNotion-care-react/commit/f4ae7c51a48f78a5589989e6d783da7134eff9c9#diff-ca30510ab0c5aa9bde117caa4700116fff90df316cfd661598130e8daa6556df)
- [x] [components/ProductFilterSidebar.js](https://github.com/nayyyhaa/inNotion-care-react/commit/f4ae7c51a48f78a5589989e6d783da7134eff9c9#diff-4d227ad9a832a4e62fdfe4ea8c885905d95c5191d9db5d0a18f469b38b286f57)
- [x] [components/ProductsListing.js](https://github.com/nayyyhaa/inNotion-care-react/commit/f4ae7c51a48f78a5589989e6d783da7134eff9c9#diff-c08da705d96013060bd9df0c0c7c7c70bdd2d34323df49be71f3b9d5f27f355e)

 #### Contexts + Reducers

- [x] [contexts/CategoriesContext.js](https://github.com/nayyyhaa/inNotion-care-react/commit/f4ae7c51a48f78a5589989e6d783da7134eff9c9#diff-06611a038634b27ab765031443d328617901f093384b491da3e49edff9a0197c)
- [x] [contexts/ProductsContext.js](https://github.com/nayyyhaa/inNotion-care-react/commit/f4ae7c51a48f78a5589989e6d783da7134eff9c9#diff-77459fb5295bf36824bc377c0f08797cc03d6fa0f5bbaf2cc4fa1e614db67b02)
- [x] [reducers/categoriesReducer.js](https://github.com/nayyyhaa/inNotion-care-react/commit/f4ae7c51a48f78a5589989e6d783da7134eff9c9#diff-9a5a29b934fb02952a452b7ba1b4ec2e858c068e1542c9159eadcb167295bf6e)
- [x] [reducers/productsReducer.js](https://github.com/nayyyhaa/inNotion-care-react/commit/f4ae7c51a48f78a5589989e6d783da7134eff9c9#diff-1e72764740711c06e4f3cf410674a7d5a3bb67e5236ee138c5019d857f828e3a)

#### utils/custom hooks:
- [x] [custom-hooks/useAsync.js](https://github.com/nayyyhaa/inNotion-care-react/commit/f4ae7c51a48f78a5589989e6d783da7134eff9c9#diff-dc778d7024e5e4ec257fd5f324180c880a16af46ec13ae5fb1f315edc5b5c3a8)
 
#### files to *ignore*
- [ ] package.json files
- [ ] pages with var name changes

### Live preview : https://deploy-preview-4--innotion-carev2.netlify.app/

### CODE REVIEW

File: src/components/FeatureCategories.js [hunk 1]
```diff
@@ -1,38 +1,41 @@
...
       </h2>
       <div className="product-container grid-3-col full-wd m-v-2 m-b-5">
-        {ProductCategoryData?.map(({ id, name, description, image, link }) => {
+        {categories?.map(({ _id, tag, categoryName, description, contentColor, image, link }) => {
           return (
             <Link
               to={link}
-              key={id}
+              key={_id}
               href="pages/products.html"
```
Comment (vrx29) [inline src/components/FeatureCategories.js]: You are using  to and href both in Link tag. I think href should not be there. In browser also when clicking on link your app is routing to html page not the route that you have defined in to={link} tag

**Review by vrx29 — CHANGES_REQUESTED** [review]

File: src/components/FeatureCategories.js (hunk 1 above)
Comment (nayyyhaa) [reply src/components/FeatureCategories.js]: okay yes missed this. I've removed the `href` from here.

File: src/components/FeatureCategories.js [hunk 2]
```diff
@@ -1,38 +1,42 @@
...
   const { dispatchFilter } = useFilter();
+  const { categories } = useCategories();
   return (
     <section className="section centered-text grid-ctr w-95p m-t-3 m-b-5">
       <h2 className="title colored-text h1 m-v-1">
         <span className="circle"></span>Because we <span className="complementary-color-text">Care</span>
       </h2>
       <div className="product-container grid-3-col full-wd m-v-2 m-b-5">
-        {ProductCategoryData?.map(({ id, name, description, image, link }) => {
+        {categories?.map(({ _id, tag, categoryName, description, contentColor, image, link }) => {
```
Comment (anik31) [inline src/components/FeatureCategories.js:14]: appreciation - good that you're using optional chaining

File: src/utils/custom-hooks/useAsync.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+export const useAsync = ({ url, actionType, dispatch, payloadType }) => {
+  useEffect(() => {
+    (async () => {
+      try {
+        const { data } = await axios(url);
+        dispatch({ type: actionType, payload: data[payloadType] });
+      } catch (err) {
+        console.log(err);
+      }
```
Comment (anik31) [inline src/utils/custom-hooks/useAsync.js:12]: appreciation - nice, you're using custom hook for data fetch

**Review by anik31 — APPROVED** [review]
Nice work, code is clean & readable 💯

**Comment by nayyyhaa** [comment]
Thanks @vrx29 & @anik31 for reviewing. I've made the said changes, now merging the PR.

**Pull request merged**